confmap/resolver.go

```diff
@@ -171,7 +171,11 @@ func (mr *Resolver) Resolve(ctx context.Context) (*Conf, error) {
 		if err != nil {
 			return nil, err
 		}
-		cfgMap[k] = val
+		if v, ok := val.(string); ok {
```

Comment:
This can also be `expandedValue`, `[]any`, and `map[string]any`. We need to handle it a  function calling itself for each of the cases

Reply:
Updated to handle new types and added more test cases to the e2e expansion tests.